### PR TITLE
nnbd annotations in flutter_goldens_client

### PR DESCRIPTION
**Pull request by a14n**
## Description

Add nnbd annotations in `packages/flutter_goldens_client`

## Related Issues

- dart-lang/sdk#40146

## Tests

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_goldens_client/lib/skia_client.dart [hunk 1]
```diff
@@ -682,9 +683,6 @@ class SkiaGoldDigest {
 
   /// Create a digest from requested JSON.
   factory SkiaGoldDigest.fromJson(Map<String, dynamic> json) {
-    if (json == null)
-      return null;
-
```
Comment (a14n) [inline packages/flutter_goldens_client/lib/skia_client.dart:687]: @Piinks: does this particular change looks good to you? I haven't found place where this constructor is call with null. 

In NNBD a factory constructor can not return `null`. If we want to return `null` the only solution is to replace this factory constructor by a static method.

Comment (Piinks) [reply packages/flutter_goldens_client/lib/skia_client.dart:687]: I think this will be ok. The only use of this factory is above on line 569. :)

**Review by Piinks — APPROVED** [review]
LGTM! Thanks! It may be useful to comment somewhere that this file is in the process of migration, just to explain all the notations. :)

**Comment by a14n** [comment]
>It may be useful to comment somewhere that this file is in the process of migration, just to explain all the notations. :)

What do you think about a `/*!*/` as comment? :D

More seriously I'm not sure it's worth to add a comment to all 2959 `*.dart` files in the migration process. The migration tool works at package level and this comment (if exists) should rather be at this level.

Finally I wonder what prevent the migration to be applied. I will give it a try.

**Pull request merged**